### PR TITLE
cli: protect against API nils

### PR DESCRIPTION
**Pull request by raybejjani**
The newer API has more nested fields and they may be nil. The CLI
tooling needs to check this more rigourously.

### CODE REVIEW

**Comment by raybejjani** [comment]
test-me-pleas

**Comment by raybejjani** [comment]
test-me-please

**Comment by joestringer** [comment]
I see there's an overlap between this PR and #3551.

**Comment by raybejjani** [comment]
Indeed. While my PR did beat the other I prefer the refactor in #3551 . We can remove the endpoint_list  changes from this one

**Comment by raybejjani** [comment]
test-me-please

File: cilium/cmd/endpoint_labels.go [hunk 1]
```diff
@@ -49,9 +49,13 @@ var endpointLabelsCmd = &cobra.Command{
 			}
 		}
 
-		if lbls, err := client.EndpointLabelsGet(id); err != nil {
+		lbls, err := client.EndpointLabelsGet(id)
+		switch {
+		case err != nil:
 			Fatalf("Cannot get endpoint labels: %s", err)
-		} else {
+		case lbls == nil || lbls.Status == nil:
```
Comment (joestringer) [inline cilium/cmd/endpoint_labels.go:56]: Is it possible for an endpoint to have no labels at all?

**Review by joestringer — APPROVED** [review]

File: cilium/cmd/endpoint_labels.go (hunk 1 above)
Comment (raybejjani) [reply cilium/cmd/endpoint_labels.go:56]: I think it would be unlikely for it to be nil, but I'm not sure what the json omitemtpty does with empty lists. I assumed the worst based on the endpoint_list case, where it seemed to be some partially generated endpoint.
tl;dr I think not but I'm a pessimist
